src/mesh/voxel_cloud.rs

```diff
@@ -538,6 +547,43 @@ impl VoxelCloud {
         // and weld naked edges
         tools::weld(&joined_voxel_mesh, (min_voxel_dimension as f32) / 4.0)
     }
+
+    #[allow(dead_code)]
+    pub fn bounding_box(&self) -> BoundingBox {
+        let voxel_dimensions = self.voxel_dimensions;
+        let block_start = self.block_start;
+        let block_end = self.block_end();
+        BoundingBox::new(
+            Point3::new(
+                block_start.x as f32 * voxel_dimensions.x,
+                block_start.y as f32 * voxel_dimensions.y,
+                block_start.z as f32 * voxel_dimensions.z,
+            ),
+            Point3::new(
+                block_end.x as f32 * voxel_dimensions.x,
+                block_end.y as f32 * voxel_dimensions.y,
+                block_end.z as f32 * voxel_dimensions.z,
+            ),
+        )
+    }
+
+    #[allow(dead_code)]
+    pub fn volume_bounding_box(&self) -> BoundingBox {
+        let voxel_dimensions = self.voxel_dimensions;
+        let (volume_start, volume_dimensions) = self.compute_volume_boundaries();
+        BoundingBox::new(
+            Point3::new(
+                (volume_start.x as f32 - 0.5) * voxel_dimensions.x,
+                (volume_start.y as f32 - 0.5) * voxel_dimensions.y,
+                (volume_start.z as f32 - 0.5) * voxel_dimensions.z,
+            ),
+            Point3::new(
+                (volume_start.x as f32 + volume_dimensions.x as f32 + 0.5) * voxel_dimensions.x,
+                (volume_start.y as f32 + volume_dimensions.x as f32 + 0.5) * voxel_dimensions.y,
+                (volume_start.z as f32 + volume_dimensions.x as f32 + 0.5) * voxel_dimensions.z,
+            ),
```

Comment:
If I understand correctly, adding and subtracting 0.5 here will increase the size of the bounding box. Imagine this situation.

```rust
let bb1 = initial_bounding_box;
let mut vc1 = VoxelCloud::from_bounding_box();
vc1.set_all(true); // We don't actually have this method, but nevermind

let bb2 = vc1.volume_bounding_box();
let vc2 = VoxelCloud::from_bounding_box();

// I think it is intuitive that this assert holds, i don't think it will for the current situation,
// because we use `floor`/`ceil` in `VoxelCloud::from_bounding_box` and we add/subtract 0.5 in `VoxelCloud::volume_bounding_box()`
assert_eq!(vc1.voxel_dimensions, vc2.voxel_dimensions);
```

Reply:
I actually realized that this also happens with our current `to_mesh` implementation. (as you noted on slack, the original behavior gives the same bounding box the mesh returned from `to_mesh` would).

So if you repeat my previous comment and go `bounding_box` -> `voxel_cloud` -> `mesh` -> `bounding_box` -> `voxel_cloud`, it will keep growing.

I am just noting this, I am not sure this is a problem. Would be a nice property if this didn't happen, though. One pretty easy fix (I think) for this would be to use `round` instead of `floor`/`ceil` in `VoxelCloud::from_bounding_box`, but not sure about how this would interact with other assumptions we have. Thoughts?

Reply:
I understand your point, however the intedned use was different. I'd prefer renaming the `volume_bounding_box` to something else and adding a method doing exactly what you expect. 

This is the intended use:
- `bounding_box()` - to compare more voxel clouds if they describe the same parts of space. downside - bounding boxes are defined in the model (world) coordinates, so there is an extra conversion. the good part - you can compare voxel clouds with voxels of various dimensions. in fact, all the upcoming voxel operations requires the voxel clouds to contain voxels of the same size, but maybe they don't have to. on the other hand, the way we'll use them, we'll always use voxels of the same size.
- `from_bounding_box()` - after comparing more bounding boxes, you can generate their intersection or union and then in return generate a new voxel cloud based on this.
- `volume_bounding_box()` is a more efficient way of `to_mesh()` -> `BoundingBox::from_mesh()`. it's not intedned to be used for more operations with bounding boxes. I'm fine with renaming this method.
- I'm not against implementing also what you suggest. It will be useful for `shrink_to_fit`, however that one is already done differently and some refactor implementing some `resize_to_bounding_box` will be required.

Reply:
Okay, let's stick with the rename for now and we can revisit later if this is painful. Some documentation would be nice, so that we don't need to relearn the problem later again. I'll add it to this PR now.